### PR TITLE
Account for using declarations in nested switch statement gotos

### PR DESCRIPTION
**Pull request by 333fred**
Fixes https://github.com/dotnet/roslyn/issues/63570. We were assuming that the label in a `goto` would always be present in our list, which wasn't true if the target was a different switch case.

### CODE REVIEW

**Comment by 333fred** [comment]
@chsienki @dotnet/roslyn-compiler for review.

**Review by chsienki — DISMISSED** [review]

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for a second review.

File: src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs:356]: >node.Label is SourceLabelSymbol { SwitchCaseLabelConstant: not null } [](http://example.com/codeflow?start=75&length=69)

Is this robust enough test? Will it handle ```default``` label? Consider adding switch labels to the map when we visit switch sections instead. The value in the map could be either a block. or a switch statement.

File: src/Compilers/CSharp/Test/Semantic/Semantics/UsingDeclarationTests.cs
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/UsingDeclarationTests.cs:610]: >goto case StringSplitOptions.None; [](http://example.com/codeflow?start=24&length=34)

Consider testing ```goto default``` as well.

**Comment by AlekseyTs** [comment]
Done with review pass (commit 1)

File: src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs
Comment (333fred) [reply src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs:356]: It will handle the `default` label, I will add a test. I originally did this suggestion, but I thought it was a bit of overkill. If you prefer, I can switch back to this approach though.

Comment (333fred) [reply src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs:356]: Actually, I guess I was wrong. I will take this approach.

**Comment by 333fred** [comment]
@AlekseyTs @chsienki for another look please.

File: src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs:13]: Is this using necessary?

Comment (333fred) [reply src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs:13]: Yes, it adds the `Deconstruct` extension method.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 3), assuming CI is passing

**Comment by 333fred** [comment]
@chsienki please take another look, thanks.

File: src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs
Comment (chsienki) [inline src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs:356]: Is this comment still correct? We won't see the case labels now right, because we `continue` above?

**Review by chsienki — APPROVED** [review]

File: src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs
Comment (333fred) [reply src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs:356]: No, we'll still see it here. We forward-define all switch case labels.

**Pull request merged**
